cmake/developer_package/compile_flags/sanitizer.cmake

```diff
@@ -5,20 +5,36 @@
 include(CheckCXXCompilerFlag)
 
 if (ENABLE_SANITIZER)
-    set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} -fsanitize=address")
-    check_cxx_compiler_flag("-fsanitize-recover=address" SANITIZE_RECOVER_ADDRESS_SUPPORTED)
-    if (SANITIZE_RECOVER_ADDRESS_SUPPORTED)
-        set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} -fsanitize-recover=address")
+    check_cxx_compiler_flag("/fsanitize=address" SANITIZE_ADDRESS_SUPPORTED)
+    if (SANITIZE_ADDRESS_SUPPORTED)
+        if (WIN32)
+            set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} /fsanitize=address")
+            check_cxx_compiler_flag("/fsanitize-recover=address" SANITIZE_RECOVER_ADDRESS_SUPPORTED)
+            if (SANITIZE_RECOVER_ADDRESS_SUPPORTED)
+                set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} /fsanitize-recover=address")
+            endif()
+            set(SANITIZER_LINKER_FLAGS "${SANITIZER_LINKER_FLAGS} /fsanitize=address")
+        else()
+            set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} -fsanitize=address")
+            check_cxx_compiler_flag("-fsanitize-recover=address" SANITIZE_RECOVER_ADDRESS_SUPPORTED)
+            if (SANITIZE_RECOVER_ADDRESS_SUPPORTED)
+                set(SANITIZER_COMPILER_FLAGS "${SANITIZER_COMPILER_FLAGS} -fsanitize-recover=address")
+            endif()
+            set(SANITIZER_LINKER_FLAGS "${SANITIZER_LINKER_FLAGS} -fsanitize=address")
+        endif()
+    else()
+        message(WARNING "Address sanitizer is not supported by current compiler. Please, check requirements: https://github.com/openvinotoolkit/openvino/wiki/AddressSanitizer-and-LeakSanitizer")
```

Comment:
It should be `message(FATAL_ERROR`. Please also wrap the line so it fits the screen. And it is OK to keep it for WIN32 only

Reply:
Updated message format in las commit